src/Microsoft.TemplateEngine.Edge/GlobalSettingsTemplatesSources/GlobalSettingsTemplatesSourcesProvider.cs

```diff
@@ -71,19 +75,22 @@ public async Task<InstallResult> InstallAsync(InstallRequest installRequest)
                     }
                 }
                 if (installersThatCanInstall.Count == 0)
-                    return InstallResult.CreateFailure("");
-                if (installersThatCanInstall.Count > 1)
-                    return InstallResult.CreateFailure("");
+                {
+                    return InstallResult.CreateFailure(InstallResult.ErrorCode.UnsupportedRequest, $"{installRequest.Identifier} cannot be installed");
+                }
+
                 var installer = installersThatCanInstall[0];
                 var installResult = await installer.InstallAsync(installRequest);
 
                 if (!installResult.Success)
+                {
                     return installResult;
+                }
 
                 var data = installer.Serialize(installResult.ManagedTemplateSource);
                 settings.SettingsLoader.GlobalSettings.Add(new Abstractions.GlobalSettings.TemplatesSourceData()
                 {
-                    Details = data.details,
+                    Details = data.details.ToDictionary(kvp => kvp.Key, kvp => kvp.Value),
```

Comment:
I guess this is intentional to create copy of dictionary? Maybe both should be IReadOnlyDictionary?